### PR TITLE
Made ShouldNotMatch with a test

### PR DESCRIPTION
**Pull request by bradarv90**
For #312

### CODE REVIEW

**Comment by bradarv90** [comment]
I'm not sure what this failed tests means. Is that related to my code or just a flakey test?

File: src/Shouldly.Tests/Strings/ShouldNotMatch.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using Shouldly.Tests.TestHelpers;
+
+namespace Shouldly.Tests.Strings
+{
+    class ShouldNotMatch : ShouldlyShouldTestScenario
```
Comment (JoeMighty) [inline src/Shouldly.Tests/Strings/ShouldNotMatch.cs]: Looks good so far, don't forget to mark the class as public.

**Comment by JoeMighty** [comment]
No, it's not you - it appears to be related to an API approval library we use. I'll try and have a look into it.

**Comment by bradarv90** [comment]
Ok. I made the class public. Thanks

**Comment by JakeGinnivan** [comment]
The API approval library is there so all public API changes have to be *approved*. This stops accidental breaking changes.

You should be able to run the test and a diff tool will pop up. You can then merge the new apis over into the .approved.cs file

File: src/Shouldly.Tests/Strings/ShouldNotMatch.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        protected override void ShouldThrowAWobbly()
+        {
+            "Cheese".ShouldNotMatch(@"\w+", "Some additional context");
+        }
+
+        protected override string ChuckedAWobblyErrorMessage
+        {
+            get
+            {
+                return "\"Cheese\" should not match \"\\w+\" but was \"Cheese\" " +
```
Comment (JakeGinnivan) [inline src/Shouldly.Tests/Strings/ShouldNotMatch.cs]: Similar to the `ShouldNotBe`, if this test fails the `but was...` will always repeat the expected.

I would just make the error `"\"Cheese\" should not match \"\\w+\" but did`

**Comment by JoeMighty** [comment]
That API Approval tool is awesome. I got the diff tool popup as you described - it's pretty slick isn't it!.

File: src/Shouldly.Tests/Strings/ShouldNotMatch.cs (hunk 2 above)
Comment (bradarv90) [inline src/Shouldly.Tests/Strings/ShouldNotMatch.cs]: I'm struggling to understand how ShouldNotBe changes the normal error message. Would you mind explaining?

Comment (JakeGinnivan) [inline src/Shouldly.Tests/Strings/ShouldNotMatch.cs]: Hey, sorry for the delay @bradarv90 

an example of the message generators is https://github.com/shouldly/shouldly/blob/master/src/Shouldly/MessageGenerators/ShouldBeMessageGenerator.cs

You could create a ShouldMatchMessageGenerator and wire it up in the same way, that then lets you customise the messages

Comment (JoeMighty) [inline src/Shouldly.Tests/Strings/ShouldNotMatch.cs]: How are you getting on with this @bradarv90? If you need any help or don't have the time to finish it off then I'm happy to finish the last bit off.

Comment (bradarv90) [inline src/Shouldly.Tests/Strings/ShouldNotMatch.cs]: I'm sorry. I've gotten so busy with school work. I think it would be best if you finished it.

Comment (JoeMighty) [inline src/Shouldly.Tests/Strings/ShouldNotMatch.cs]: No problem! Thanks for all you've done :)

**Comment by JoeMighty** [comment]
As @bradarv90 wasn't able to finish this one off I finished it as part of pr329 by adding the ShouldNotMatchMessageGenerator.

**Pull request merged**